sirepo/pkcli/test_http.py

```diff
@@ -144,68 +195,59 @@ async def get(self, uri):
             )
 
     async def login(self):
-        r = await self.post('/simulation-list', PKDict())
+        r = await self.post('/simulation-list', PKDict(), self)
         assert r.srException.routeName == 'missingCookies'
-        r = await self.post('/simulation-list', PKDict())
+        r = await self.post('/simulation-list', PKDict(), self)
         assert r.srException.routeName == 'login'
-        async with self._global_lock:
-            self._login_locks.pksetdefault(self.email, asyncio.Lock())
-        async with self._login_locks[self.email]:
-            r = await self.post('/auth-email-login', PKDict(email=self.email))
-            t = sirepo.util.create_token(
-                self.email,
-            ).decode()
+        r = await self.post('/auth-email-login', PKDict(email=self.email), self)
+        t = sirepo.util.create_token(
+            self.email,
+        )
+        r = await self.post(
+            self._uri('/auth-email-authorized/{}/{}'.format(self.sim_type, t)),
+            PKDict(token=t, email=self.email),
+            self,
+        )
+        assert r.state != 'srException', 'r={}'.format(r)
+        if r.authState.needCompleteRegistration:
             r = await self.post(
-                self._uri('/auth-email-authorized/{}/{}'.format(self.sim_type, t)),
-                data=PKDict(token=t, email=self.email),
+                '/auth-complete-registration',
+                PKDict(displayName=self.email),
+                self,
             )
-            assert r.state != 'srException', 'r={}'.format(r)
-            if r.authState.needCompleteRegistration:
-                r = await self.post(
-                    '/auth-complete-registration',
-                    PKDict(displayName=self.email),
-                )
-        r = await self.post('/simulation-list', PKDict())
-        self._sid = PKDict([(x.name, x.simulationId) for x in r])
-        self._sim_db = PKDict()
-        self._sim_data = sirepo.sim_data.get_class(self.sim_type)
         return self
 
     def parse_response(self, resp):
-        self.resp = resp
-        assert self.resp.code == 200, 'resp={}'.format(resp)
-        self.json = None
+        assert resp.code == 200, 'resp={}'.format(resp)
         if 'Set-Cookie' in resp.headers:
             self._headers.Cookie = resp.headers['Set-Cookie']
         if 'json' in resp.headers['content-type']:
-            self.json = pkjson.load_any(resp.body)
-            return self.json
+            return pkjson.load_any(resp.body)
         try:
-            b = resp.body.decode()
+            b = pkcompat.from_bytes(resp.body)
         except UnicodeDecodeError:
             # Binary data files can't be decoded
```

Comment:
This is subtle. I am not sure we want to detect this case. Rather, we should know what we are expecting and assert that. If we are expecting binary data, pass that in to parse_response.